src/main/java/vimification/model/task/Task.java

```diff
@@ -41,6 +46,17 @@ public boolean containsKeyword(String keyword) {
         return description.contains(keyword);
     }
 
+    public void addTag(String newTag) {
+        newTag = newTag.toLowerCase();
+        tags.add(newTag);
+    }
+
+    public void removeTag(String tag) {
+        if (!tags.remove(tag)) {
+            throw new IllegalArgumentException("Tag does not exist");
+        }
+    }
+
     public abstract Task clone();
 
     public boolean isSameTask(Task otherTask) {
```

Comment:
???

Reply:
isSameTask() is a more lenient method than equals(), as it only compares the task description. It may be a redundant method as it is a legacy code from AB3 (used to make sure no 2 tasks can have the same description).